src/tests/jsonLogic.test.js

```diff
@@ -98,6 +103,50 @@ describe('cross-value validations', () => {
       console.error.mockRestore();
     });
 
+    it('Should throw when a var does not exist in a rule.', () => {
+      createHeadlessForm(schemaWithVarThatDoesNotExist, { strictInputType: false });
+      expect(console.error).toHaveBeenCalledWith(
+        'JSON Schema invalid!',
+        Error('"field_b" in rule "a_greater_than_ten" does not exist as a JSON schema property.')
+      );
+    });
+
+    it('Should throw when a var does not exist in a deeply nested rule', () => {
+      createHeadlessForm(schemaWithDeepVarThatDoesNotExist, { strictInputType: false });
+      expect(console.error).toHaveBeenCalledWith(
+        'JSON Schema invalid!',
+        Error('"field_b" in rule "a_greater_than_ten" does not exist as a JSON schema property.')
+      );
+    });
+
+    it('Should throw when a var does not exist in a fieldset.', () => {
+      createHeadlessForm(schemaWithDeepVarThatDoesNotExistOnFieldset, { strictInputType: false });
+      expect(console.error).toHaveBeenCalledWith(
+        'JSON Schema invalid!',
+        Error('"field_a" in rule "a_greater_than_ten" does not exist as a JSON schema property.')
+      );
+    });
+
+    it('On a property, it should throw an error for a requiredValidation that does not exist', () => {
+      createHeadlessForm(schemaWithValidationThatDoesNotExistOnProperty, {
+        strictInputType: false,
+      });
+      expect(console.error).toHaveBeenCalledWith(
+        'JSON Schema invalid!',
+        Error(`Validation "iDontExist" required for "field_a" doesn't exist.`)
+      );
+    });
+
+    it('A top level logic keyword will not be able to reference fieldset properties', () => {
+      createHeadlessForm(schemaWithPropertyThatDoesNotExistInThatLevelButDoesInFieldset, {
+        strictInputType: false,
+      });
+      expect(console.error).toHaveBeenCalledWith(
+        'JSON Schema invalid!',
+        Error('"child" in rule "validation_parent" does not exist as a JSON schema property.')
+      );
+    });
+
     it('Should throw when theres a missing rule', () => {
       createHeadlessForm(schemaWithMissingRule, { strictInputType: false });
       expect(console.error).toHaveBeenCalledWith(
```

Comment:
Fantastic work here with the error handling! I was wondering if we should throw an error when a rule is invalid.

Like: 

```json
"rule": {
  "++": [{ "var": "field_a" }, { "var": "field_b" }]
}
```

Reply:
can't believe i didnt think of this. excellent suggestion

Reply:
note here: json-logic already does do this to a degree, but meh error handling. So I'm trying to solve it as part of the rule integrity checks (https://github.com/remoteoss/json-schema-form/pull/39/commits/d03b369e958ade8b1d3fe0455c7461fcf79ac4cc), what do you think?

Reply:
I think it's ok, thanks! Just one note, looking at the [json-logic-js source code](https://github.com/jwadhams/json-logic-js/blob/master/logic.js#L369), it can throw a different error message when dealing with sub rules. But Probably we don't need to deal with it for now.